sources/packages/web/src/views/student/financial-aid-application/StudentApplication.vue

```diff
@@ -1,103 +1,151 @@
 import { ApplicationService } from "@/services/ApplicationService";
 import { StudentRoutesConst } from "@/constants/routes/RouteConstants";
 import { ProgramYearService } from "@/services/ProgramYearService";
+import ContentGroup from "@/components/generic/ContentGroup.vue";
 
-export default {
-  components: { ModalDialogBase },
+export default defineComponent({
+  components: { ModalDialogBase, ContentGroup },
   setup() {
     const initialData = ref({});
     const router = useRouter();
     const snackBar = useSnackBar();
-    const formioDataLoader = useFormioDropdownLoader();
-    const formioUtils = useFormioUtils();
     const { showDialog, showModal } = useModalDialog<void>();
-    const PROGRAM_YEAR_DROPDOWN_KEY = "programYear";
+    const programYearOptions = ref([] as SelectItemType[]);
+    const programYearId = ref();
+    const startApplicationForm = ref({} as VForm);
+    const { checkNullOrEmptyRule } = useRules();
 
-    const formLoaded = async (form: any) => {
-      await formioDataLoader.loadProgramYear(form, PROGRAM_YEAR_DROPDOWN_KEY);
-    };
-    const dialogClosed = () => {
+    onMounted(async () => {
+      const programYearOptionTmp =
+        await ProgramYearService.shared.getProgramYearOptions();
+      programYearOptions.value = programYearOptionTmp.map((yearOption) => {
+        return {
+          title: yearOption.description,
+          value: yearOption.id.toString(),
+        };
+      });
+    });
+
+    const closeDialog = () => {
       showDialog.value = false;
     };
-    const customEventCallback = async (form: any, event: FormIOCustomEvent) => {
+    const startApplication = async () => {
       try {
-        const programYearId = formioUtils.getComponentValueByKey(
-          form,
-          PROGRAM_YEAR_DROPDOWN_KEY,
-        );
-        const createDraftResult =
-          await ApplicationService.shared.createApplicationDraft({
-            programYearId: programYearId,
-            data: {},
-            associatedFiles: [],
-          });
-        if (createDraftResult.draftAlreadyExists) {
-          showDialog.value = true;
+        const validationResult = await startApplicationForm.value.validate();
+        if (!validationResult.valid) {
           return;
         }
-        const programYear =
-          await ProgramYearService.shared.getActiveProgramYear(programYearId);
-        if (
-          programYear &&
-          programYear.formName &&
-          FormIOCustomEventTypes.RouteToStartFullTimePartTimeApplication ===
-            event.type
-        ) {
-          router.push({
-            name: StudentRoutesConst.DYNAMIC_FINANCIAL_APP_FORM,
-            params: {
-              selectedForm: programYear.formName,
-              programYearId: programYearId,
-              id: Number(createDraftResult.draftId),
-            },
-          });
+        if (programYearId.value) {
+          const programYearNumberId = Number(programYearId.value);
```

Comment:
if we define the type of `programYearId` do we need this?

Reply:
Yeah, I can change that to be typed. Thanks!